### PR TITLE
Fixes #12789: Add slackware support to rudder-packages

### PR DESCRIPTION
**Pull request by VictorQrt**
https://www.rudder-project.org/redmine/issues/12789

### CODE REVIEW

**Comment by VictorQrt** [comment]
Commit modified

**Comment by VictorQrt** [comment]
Commit modified

**Comment by VictorQrt** [comment]
Commit modified

File: rudder-agent/SOURCES/rudder-agent-postinst [hunk 1]
```diff
@@ -33,6 +33,9 @@ then
     /usr/bin/mkssys -s rudder-agent -p "${CFE_DIR}/bin/cf-execd" -a "-F" -u root -S -n15 -f9 -R
     /usr/sbin/mkitab "rudder-agent:23456789:once:/usr/bin/startsrc -s rudder-agent"
     # No need to tell init to re-read /etc/inittab, it does it automatically every 60 seconds
+  elif [ ${CFRUDDER_OS} = "slack" ]
+  then
+    # TODO
```
Comment (peckpeck) [inline rudder-agent/SOURCES/rudder-agent-postinst]: This is not finished

Comment (peckpeck) [reply rudder-agent/SOURCES/rudder-agent-postinst]: you need a command here
example : true

File: rudder-agent/slackware/doinst.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+if [ -L /etc/rc.d/rc.rudder-agent ]; then
+	# upgrade
+	/opt/rudder/share/package-scripts/rudder-agent-postinst 0 slack false
```
Comment (peckpeck) [inline rudder-agent/slackware/doinst.sh]: 0 => false

File: rudder-agent/slackware/doinst.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+	if [ "$(readlink -f /etc/rc.d/rc.rudder-agent)" != "/etc/init.d/rudder-agent" ]; then
+		unlink /etc/rc.d/rc.rudder-agent
+		ln -s /etc/init.d/rudder-agent /etc/rc.d/rc.rudder-agent
+	fi
+else
+	# new install
+	/opt/rudder/share/package-scripts/rudder-agent-postinst 1 slack false
```
Comment (peckpeck) [inline rudder-agent/slackware/doinst.sh]: 1 => true

File: rudder-agent/slackware/doinst.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
+if [ -L /etc/rc.d/rc.rudder-agent ]; then
+	# upgrade
+	/opt/rudder/share/package-scripts/rudder-agent-postinst 0 slack false
+	if [ "$(readlink -f /etc/rc.d/rc.rudder-agent)" != "/etc/init.d/rudder-agent" ]; then
```
Comment (peckpeck) [inline rudder-agent/slackware/doinst.sh]: when would that happen ?

File: rudder-agent/slackware/doinst.sh [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+		unlink /etc/rc.d/rc.rudder-agent
+		ln -s /etc/init.d/rudder-agent /etc/rc.d/rc.rudder-agent
+	fi
+else
+	# new install
+	/opt/rudder/share/package-scripts/rudder-agent-postinst 1 slack false
+	ln -s /etc/init.d/rudder-agent /etc/rc.d/rc.rudder-agent
```
Comment (peckpeck) [inline rudder-agent/slackware/doinst.sh]: Maybe put this on the line before

File: rudder-agent/slackware/rudder-agent-upgrade-slack [hunk 6]
```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
```
Comment (peckpeck) [inline rudder-agent/slackware/rudder-agent-upgrade-slack]: Remove

File: rudder-agent/slackware/rudder-agent.SlackBuild [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+#
+# Contributors : Jeremy HOCDE
+#                Victor QUERETTE
+#
+#####################################################################################
+
+# Slackware build script for rudder-agent
+
+PRGNAM=rudder-agent
+VERSION=${VERSION:-4.3.1}
```
Comment (peckpeck) [inline rudder-agent/slackware/rudder-agent.SlackBuild]: We need to think about when we create the build task

File: rudder-agent/slackware/rudder-agent.SlackBuild [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+cd ../SOURCES
+env="USE_PAM=false RUDDER_VERSION_TO_PACKAGE=$VERSION DESTDIR=$PKG/ USE_SYSTEM_PERL=false"
+make $env -j6
+make install $env -j6
+cd $PKG
+
+# Strip binaries and libraries - this can be done with 'make install-strip'
+# in many source trees, and that's usually acceptable if so, but if not,
+# use this:
+find $PKG -print0 | xargs -0 file | grep -e "executable" -e "shared object" | grep ELF \
```
Comment (peckpeck) [inline rudder-agent/slackware/rudder-agent.SlackBuild]: Check that we don't already do this in the Makefile

File: rudder-agent/slackware/rudder-agent.SlackBuild [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+  | cut -f 1 -d : | xargs strip --strip-unneeded 2> /dev/null || true
+
+# Compress man pages
+# If the man pages are installed to /usr/share/man instead, you'll need to either
+# add the --mandir=/usr/man flag to configure or move them manually after the
+# make install process is run.
+find $PKG/opt/rudder/share/man -type f -exec gzip -9 {} \;
```
Comment (peckpeck) [inline rudder-agent/slackware/rudder-agent.SlackBuild]: Idem

File: rudder-agent/slackware/rudder-agent.SlackBuild [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+for i in $( find $PKG/opt/rudder/share/man -type l ) ; do ln -s $( readlink $i ).gz $i.gz ; rm $i ; done
+
+# Remove perllocal.pod and other special files that don't need to be installed,
+# as they will overwrite what's already on the system.  If this is not needed,
+# remove it from the script.
+# Remove 'special' files
+find $PKG -name perllocal.pod \
```
Comment (peckpeck) [inline rudder-agent/slackware/rudder-agent.SlackBuild]: Idem

File: rudder-agent/slackware/rudder-agent.SlackBuild [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+  | xargs rm -f
+
+# Copy program documentation into the package
+# The included documentation varies from one application to another, so be sure
+# to adjust your script as needed
+# Also, include the SlackBuild script in the documentation directory
+mkdir -p $PKG/usr/doc/$PRGNAM-$VERSION
+cp -a \
+  $CWD/../SOURCES/rudder-sources/rudder-doc/* \
+  $PKG/usr/doc/$PRGNAM-$VERSION
```
Comment (peckpeck) [inline rudder-agent/slackware/rudder-agent.SlackBuild]: idem

**Comment by VictorQrt** [comment]
Commit modified

**Comment by VictorQrt** [comment]
Commit modified

**Pull request merged**